iotfunctions/pipeline.py

```diff
@@ -2341,6 +2342,31 @@ def _execute_stage(self, stage, df, start_ts, end_ts, entities, register, to_csv
         self.trace_add(msg, created_by=stage, df=newdf)
         return newdf
 
+    def apply_scope(self, df, stage):
+        if hasattr(stage, 'scope') and stage.scope is not None:
+            logger.debug('Applying Scope')
+            eval_expression = ''
+            scope = json.loads(stage.scope)
+            if scope.get('type') == 'DIMENSIONS':
+                logger.debug('Applying Dimensions Scope')
+                dimension_count = len(scope.get('dimensions'))
+                for dimension_filter in scope.get('dimensions'):
```

Comment:
you can do length check outside of the for loop.

Reply:
Not clear what you meant